### PR TITLE
[Enhancement] Pending SO Items for Purchase Request Report 

### PR DESCRIPTION
**Pull request by Mangesh-Khairnar**
- Pending Sales Order Items for Purchase Request Report which was written as a query report didn't cover all the edge cases and was slower to read for larger datasets. This report refactor intends to solve that problem. 
- Sales Order for which a material request was  partially completed will only have remaining items listed and those sales order which don't have a corresponding material request will also be listed(For SO which are not "Completed").

pendingsoitemsforpurchasereport

### CODE REVIEW

**Review by rmehta — CHANGES_REQUESTED** [review]
please also add comments and a test case also a better name could be:

"Pending Sales Items to be Purchased"

File: erpnext/selling/report/pending_so_items_for_purchase_request/pending_so_items_for_purchase_request.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+			"label": _("Company"),
+			"fieldname": "company",
+			"fieldtype": "Data",
+			"width": 140
+		}
+	]
+	return columns
+
+def get_data():
+	sales_order_entry = frappe.db.sql("""select  so_item.item_code,
```
[the remark this answers is not in the document]
Comment (Mangesh-Khairnar) [reply erpnext/selling/report/pending_so_items_for_purchase_request/pending_so_items_for_purchase_request.py]: added indentation, comments and required tests

**Comment by djpalshikar** [comment]
@Mangesh-Khairnar, is this going to cover items for Pending for  **Purchase** Only.
 
We manufacture our items, therefore we create Material Request for Manufacture.

**Comment by Mangesh-Khairnar** [comment]
@djpalshikar this tracks Sales Order that have partial or no Material Request created. It has the same functionality as requested on issue #14953.

**Comment by djpalshikar** [comment]
@Mangesh-Khairnar , in that case should not be renamed  as "Pending SO Items", as this will cover both items to be Purchased as well as items to be manufactured.

**Pull request merged**